spring-integration-core/src/main/java/org/springframework/integration/config/PublisherRegistrar.java

```diff
@@ -46,10 +46,8 @@
 	public void registerBeanDefinitions(AnnotationMetadata importingClassMetadata, BeanDefinitionRegistry registry) {
 		Map<String, Object> annotationAttributes =
 				importingClassMetadata.getAnnotationAttributes(EnablePublisher.class.getName());
-		if (annotationAttributes == null) {
-			return;
-		}
-		String value = (String) annotationAttributes.get("value");
+
+		String value = (annotationAttributes == null ? null : (String) annotationAttributes.get("value"));
```

Comment:
Perhaps it would be better to do `AnnotationUtils.getDefaultValue(EnablePublisher.class)` - this would protect against annotation changes in the future.